contrib/database/sql/conn.go

```diff
@@ -253,7 +253,26 @@ func (tc *TracedConn) injectComments(ctx context.Context, query string, mode tra
 	if span, ok := tracer.SpanFromContext(ctx); ok {
 		spanCtx = span.Context()
 	}
-	carrier := tracer.SQLCommentCarrier{Query: query, Mode: mode, DBServiceName: tc.cfg.serviceName, PeerDBHostname: tc.meta[ext.TargetHost], PeerDBName: tc.meta[ext.DBName]}
+
+	// This occurs if the user sets peer.service explicitly while creating the connection
+	// through the use of sqltrace.WithSpanTags
+	var peerService string
+	if meta, ok := ctx.Value(spanTagsKey).(map[string]string); ok {
+		if peerServiceTag, ok := meta[ext.PeerService]; ok {
+			peerService = peerServiceTag
+		}
+	}
+
+	// This occurs if the SQL Connection is opened or registered using
+	// WithCustomTags.  This is lower precedence than above since it is
+	// less specific
+	if peerService == "" && len(tc.cfg.tags) > 0 {
+		if v, ok := tc.cfg.tags[ext.PeerService].(string); ok {
+			peerService = v
+		}
+	}
```

Comment:
I think this code could benefit from being refactored into a new function, since now there are some precendence rules and might not be so obvious what is going on in the future (also the code is easier to follow using `return`'s instead of variable reassignments):

 ```go
// providedPeerService returns the user provided peer.service from the different sources it can be specified.
func (tc *TracedConn) providedPeerService(ctx context.Context) string {
	// This occurs if the user sets peer.service in the db operation context
	// through the use of sqltrace.WithSpanTags
	if meta, ok := ctx.Value(spanTagsKey).(map[string]string); ok {
		if val, ok := meta[ext.PeerService]; ok {
			return val
		}
	}

	// This occurs if the SQL Connection is opened or registered using
	// sqltrace.WithCustomTags. This is lower precedence than above since it is
	// less specific
	if tp.cfg.tags != nil {
		if val, ok := tc.cfg.tags[ext.PeerService].(string); ok {
			return val
		}
	}
	return ""
}
```